examples/complex-chart/complex-chart-example.js

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React from 'react';
+
+import {
+  XYPlot,
+  XAxis,
+  YAxis,
+  HorizontalGridLines,
+  makeWidthFlexible,
+  LineSeries,
+  VerticalBarSeries,
+  DiscreteColorLegend,
+  Crosshair
+} from 'react-vis';
+
+const FlexibleXYPlot = makeWidthFlexible(XYPlot);
+
+/**
+ * Get the array of x and y pairs.
+ * The function tries to avoid too large changes of the chart.
+ * @param {number} total Total number of values.
+ * @returns {Array} Array of data.
+ * @private
+ */
+function getRandomSeriesData(total) {
+  const result = [];
+  let lastY = Math.random() * 40 - 20;
+  let y;
+  const firstY = lastY;
+  for (let i = 0; i < total; i++) {
+    y = Math.random() * firstY - firstY / 2 + lastY;
+    result.push({
+      x: i,
+      y
+    });
+    lastY = y;
+  }
+  return result;
+}
+
+export default class Example extends React.Component {
+
+  constructor(props) {
+    super(props);
+    const totalValues = Math.random() * 50;
+    this.state = {
+      crosshairValues: [],
+      series: [
+        {
+          title: 'Apples',
+          disabled: false,
+          data: getRandomSeriesData(totalValues)
+        },
+        {
+          title: 'Bananas',
+          disabled: false,
+          data: getRandomSeriesData(totalValues)
+        }
+      ]
+    };
+    this._nearestXHandler = this._nearestXHandler.bind(this);
```

Comment:
If you use class properties for your methods, you won't have to bind all of them in the constructor, much cleaner.